src/main/ruby/truffleruby/core/truffle/range_operations.rb

```diff
@@ -79,11 +79,10 @@ def self.range_cover?(range, other)
       other_e = other.end
 
       return false if !Primitive.nil?(e) && Primitive.nil?(other_e)
-
-      return false if !Primitive.nil?(other_e) &&
+      return false if !Primitive.nil?(range.begin) && Primitive.nil?(other_b)
+      return false if !Primitive.nil?(other_b) && !Primitive.nil?(other_e) &&
           range_less(other_b, other_e) > (other.exclude_end? ? -1 : 0)
-
-      return false unless cover?(range, other_b)
+      return false if !Primitive.nil?(other_b) && !cover?(range, other_b)
```

Comment:
This mostly isn't your doing, but I find this `cover_range?` logic incredibly difficult to read for something that should essentially be very simple.

Is there any reason we couldn't rewrite the whole thing as:

```ruby
def self.range_cover?(range, other)
  b1 = range.begin
  b2 = other.begin
  e1 = range.end
  e2 = other.end

  return false if Primitive.nil?(b2) && !Primitive.nil?(b1)
  return false if Primitive.nil?(e2) && !Primitive.nil?(e1)

  (Primitive.nil?(b2) || cover?(range, b2)) && (Primitive.nil?(e2) || cover?(range, e2))
end
```

Reply:
That looks so much better, thank you! Slight modifications due to exclude_end? things.